stakingModel/contracts/ContributionCalculation.sol

```diff
@@ -0,0 +1,67 @@
+pragma solidity 0.5.4;
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "@daostack/arc/contracts/controller/Avatar.sol";
+import "../../contracts/dao/schemes/SchemeGuard.sol";
+import "../../contracts/DSMath.sol";
+import "../../contracts/token/GoodDollar.sol";
+
+/* @title Contribution calculation for selling gd tokens
+ */
+contract ContributionCalculation is DSMath, SchemeGuard {
+    using SafeMath for uint256;
+
+    GoodDollar gooddollar;
+    uint256 public sellContributionRatio;
+
+    event SellContributionRatioUpdated(address indexed caller,
+                                       uint256 nom,
+                                       uint256 denom);
+
+    constructor(
+        Avatar _avatar,
+        address _gooddollar,
```

Comment:
it should know also reserve/market maker so it can use parameters from there, like current reserve ratio,
total in reserve etc...

Reply:
added in new functions ea7fa3b9d47af8ec3aead553cd095899c8c4da66